examples/parse-csv-lists/parse_csv_lists_to_ballerina_types.bal

```diff
@@ -0,0 +1,44 @@
+import ballerina/data.csv;
+import ballerina/io;
+
+// Represents the details of an employee.
+type Employee record {|
+    int empId;
+    string empName;
+    string department;
+    float salary;
+|};
+
+// Represents the salary details of an employee.
+type EmployeeSalary record {|
+    // This annotation is used to map the empId field in the source CSV record 
+    // to the id field in the target record.
+    @csv:Name {
+        value: "empId"
+    }
+    int id;
+    float salary;
+|};
+
+public function main() returns error? {
+    // Sample CSV string array representing employee data.
+    string[][] csvList = [
+        ["1", "John", "Engineering", "1000.0"],
+        ["2", "Doe", "HR", "2000.0"],
+        ["3", "Jane", "Finance", "3000.0"]
+    ];
+
+    // Parse the CSV list into an array of `Employee` records by specifying custom headers.
+    Employee[] employees = check csv:parseList(csvList, {customHeaders: ["empId", "empName", "department", "salary"]});
```

Comment:
Is there a reason why we `check` error in this PR and in some others (#5668,  #5669, etc. ) we take both values? (I  personally prefer `check` unless you are doing something with errors)

Reply:
Yeah, using `check` will be the better option

Reply:
Updated